src/substrait/proto/CMakeLists.txt

```diff
@@ -79,12 +79,23 @@ endforeach()
 # Add the generated protobuf C++ files to our exported library.
 add_library(substrait_proto ${PROTO_SRCS} ${PROTO_HDRS} ProtoUtils.cpp
                             ProtoUtils.h)
+target_sources(
+  substrait_proto PUBLIC FILE_SET HEADERS BASE_DIRS
+                         ${PROTO_OUTPUT_TOPLEVEL_DIR} FILES ${PROTO_HDRS})
```

Comment:
This is a mechanism introduced in CMake v2.23. I found this [SO post](https://stackoverflow.com/questions/54271925/how-to-configure-cmakelists-txt-to-install-public-headers-of-a-shared-library) quite useful. Not sure if we want this. If yes, the CMake version requirements in the top-level CMake file should probably be changed. Otherwise, we could have one `install(FILES ...)` command per file in the loop above.